easier68k/core/opcodes/cmp.py

```diff
@@ -95,31 +95,35 @@ def execute(self, simulator: M68K):
         src_val = self.src.get_value(simulator, self.size.get_number_of_bytes())
         dest_val = self.dest.get_value(simulator, self.size.get_number_of_bytes())
 
-        comparison = src_val.get_value_signed() - dest_val.get_value_signed()
+        comparison = dest_val.get_value_signed() - src_val.get_value_signed()
+        raw_total = dest_val.get_value_unsigned() - src_val.get_value_unsigned()
         comp_mv = dest_val - src_val
 
         # mask out only the bits we need/want
-        result_mv_masked = MemoryValue(self.size,
-                                       unsigned_int=mask_value_for_length(self.size, comp_mv.get_value_unsigned()))
+        comp_mv = MemoryValue(self.size,
+                              unsigned_int=mask_value_for_length(self.size, comp_mv.get_value_unsigned()))
+
+        negative = comp_mv.get_negative()
+
         overflow = False
-        # if the two values have different MSB (in the size of this opcode)
-        if src_val.get_msb(self.size) != dest_val.get_msb(self.size):
-            # and after subtracting have a different MSB
-            # then there has been an overflow
-            if src_val.get_negative() != result_mv_masked.get_negative():
-                overflow = True
+
+        # src false
+        if src_val.get_msb(self.size) is True:
+            if dest_val.get_msb(self.size) is False:
+                if raw_total > 0 and raw_total & 0x80000000 > 0:
+                    overflow = True
```

Comment:
This logic was taken from CMPI, which was taken from SUB and SUBQ. However, I used new logic to identify the overflow bit for SUB and SUBQ. I would recommend checking it out and seeing if it might work better than this implementation.